packages/vite/src/node/optimizer/scan.ts

```diff
@@ -294,8 +294,23 @@ async function prepareEsbuildScanner(
     const { tsconfig } = await loadTsconfigJsonForFile(
       path.join(environment.config.root, '_dummy.js'),
     )
-    if (tsconfig.compilerOptions?.experimentalDecorators) {
-      tsconfigRaw = { compilerOptions: { experimentalDecorators: true } }
+    if (
+      tsconfig.compilerOptions?.experimentalDecorators ||
```

Comment:
I'd imagine there was a reason to only pick up `experimentalDecorators` instead of entire `tsconfig`, but I didn't understand the comment above and didn't have tests failing with expanding jsx here too.

Reply:
Setting jsx related fields is ok.
The comment mainly explains the behavior difference between the scanner (uses the closest tsconfig.json from **the root**) and the plugin container (uses the closest tsconfig from **each file**).
Fields other than `experimentalDecorators` were not set just because it didn't cause any syntax errors.